src/modules/test/README.md

```diff
@@ -1,60 +1,98 @@
+  }
+]
+
+```
+
+### Command steps
+
+Command steps are used to execute a command as part of the test suite. Success or failure of the command is determined by the exit code of the command. By default, an exit code of `0` is considered a success, however a `Status` can be provided to override this behavior.
+
+> Command steps are useful for verifying the state of a device before/after a desired property is changed and adding information to the test logs. (e.g. `grep` the contents of a file to verify that a desired property was set correctly.)
+
+> A `Delay` can also be added to a command step to allow for processing from the previous step.
+
+```json
+[
+  {
+    "RunCommand": "grep -q 'my-network-interface' /etc/wpa_supplicant/wpa_supplicant.conf"
   },
   {
-    "ComponentName": "",
-    "ObjectName": "",
-    "Desired": false,
-    "ExpectedResult": 22,
-    "WaitSeconds": 0
+    "RunCommand": "exit 123",
+    "Status": 123
   }
 ]
 ```
 
-See [Googletest Primer - Simple Tests](https://google.github.io/googletest/primer.html#simple-tests) for more details on test suites and test naming.
-
-# Usage
-1. `modulestest` or `modulestest testplate.json` (equivalent) tests all modules which have [test recipe](https://github.com/Azure/azure-osconfig/tree/main/src/modules/test/recipes) and [MIM](https://github.com/Azure/azure-osconfig/tree/main/src/modules/mim) defined.
-2. `moduletest <module name>` Example: `modulestest tpm` will perform the [test recipe](https://github.com/Azure/azure-osconfig/tree/main/src/modules/test/recipes) defined for the _tpm_ module.
-3. `moduletest <module.so>` Example: `modulestest /azure-osconfig/build/modules/tpm/src/so/tpm.so` performs a basic test on the module only, this ensures the MMI interface is properly defined. No payloads are sent to modules, we simply load the module, ensure _MMIGetInfo_ returns correctly.
-4. `moduletest <module.so> <mim.json> <recipe.json>` Example: `modulestest "/azure-osconfig/build/modules/tpm/src/so/tpm.so" "/azure-osconfig/src/modules/mim/tpm.json" "/azure-osconfig/src/modules/test/recipes/TpmTests.json"` Equivalent to #2
+### Module steps
 
-Supports any of the `-gtest_*` flags. See `--help` for additional usage details.
- * `--gtest_list_tests`
- * `--gtest_filter=POSTIVE_PATTERNS[-NEGATIVE_PATTERNS]`
+Module steps are used to load and unload a module as part of the test suite. This can be useful for testing the behavior of a module when it is loaded/unloaded and gives the opportunity to verify the state of the device before/after the module is loaded/unloaded.
 
-# Authoring Test Recipes
-See *Testing* in [docs/modules.md](https://github.com/Azure/azure-osconfig/blob/main/docs/modules.md#13-testing) for more information on authoring Test Recipes (see [src/modules/test/recipes/](https://github.com/Azure/azure-osconfig/tree/main/src/modules/test/recipes) for current recipes).
+> If the unload step is omitted, the module will be unloaded after the test suite is complete.

```

Comment:
>will be unloaded after the test suite is complete. [](http://example.com/codeflow?start=44&length=50)

Why not leave it loaded until ModuleTest process exits?

Maybe in the future some module developers would like to see if their module remains loaded and does not crash while other modules are in use.

Reply:
Agreed, fixed the log trace to display a warning to make this behavior more clear.